docs/sphinx/pypi-dependencies.md

```diff
@@ -0,0 +1,153 @@
+call it to create the central external repo and individual wheel external repos.
+
+```
+load("@rules_python//python:pip.bzl", "pip_parse")
+
+# Create a central repo that knows about the dependencies needed from
+# requirements_lock.txt.
+pip_parse(
+   name = "my_deps",
+   requirements_lock = "//path/to:requirements_lock.txt",
+)
+# Load the starlark macro, which will define your dependencies.
+load("@my_deps//:requirements.bzl", "install_deps")
+# Call it to define repos for your requirements.
+install_deps()
+```
+
+#### pip rules
+
+Note that since `pip_parse` is a repository rule and therefore executes pip at
+WORKSPACE-evaluation time, Bazel has no information about the Python toolchain
+and cannot enforce that the interpreter used to invoke pip matches the
+interpreter used to run `py_binary` targets. By default, `pip_parse` uses the
+system command `"python3"`. To override this, pass in the `python_interpreter`
+attribute or `python_interpreter_target` attribute to `pip_parse`.
+
+You can have multiple `pip_parse`s in the same workspace.  Or use the pip
+extension multiple times when using bzlmod. This configuration will create
+multiple external repos that have no relation to one another and may result in
+downloading the same wheels numerous times.
+
+As with any repository rule, if you would like to ensure that `pip_parse` is
+re-executed to pick up a non-hermetic change to your environment (e.g., updating
+your system `python` interpreter), you can force it to re-execute by running
+`bazel sync --only [pip_parse name]`.
+
+:::{note}
+The `pip_install` rule is deprecated. `pip_parse` offers identical
+functionality, and both `pip_install` and `pip_parse` now have the same
+implementation. The name `pip_install` may be removed in a future version of the
+rules.
+:::
+
+The maintainers have made all reasonable efforts to facilitate a smooth
+transition. Still, some users of `pip_install` will need to replace their
+existing `requirements.txt` with a fully resolved set of dependencies using a
+tool such as `pip-tools` or the `compile_pip_requirements` repository rule.
+
+### Using third_party packages as dependencies
+
+Each extracted wheel repo contains a `py_library` target representing
+the wheel's contents. There are two ways to access this library. The
+first uses the `requirement()` function defined in the central
+repo's `//:requirements.bzl` file. This function maps a pip package
+name to a label:
+
+```
+load("@my_deps//:requirements.bzl", "requirement")
+
+py_library(
+    name = "mylib",
+    srcs = ["mylib.py"],
+    deps = [
+        ":myotherlib",
+        requirement("some_pip_dep"),
+        requirement("another_pip_dep"),
+    ]
+)
+```
+
+The reason `requirement()` exists is that the pattern for the labels,
+while not expected to change frequently, is not guaranteed to be
+stable. Using `requirement()` ensures you do not have to refactor
+your `BUILD` files if the pattern changes.
```

Comment:
nit: this is an interesting statement. not sure if we can actually rely on this in the future when making changes to it though.